frontend/src/components/object/ObjectTable.vue

```diff
@@ -98,6 +131,50 @@ const filters = ref({
   // @ts-ignore
   global: { value: null, matchMode: FilterMatchMode.CONTAINS }
 });
+
+const onSelectAllChange = (event: any) => {
+  selectAll.value = event.checked;
+  if (event.checked) {
+    if (isInitialLoad.value) {
+      /*
+       * On initial page load, the object order returned by this emit doesn't corespond
+       * to the actual display order - it's just the order returned by the COMS API.
+       * Workaround: directly query the DOM to figure out the actual display order (see queryTableOrder()).
+       */
+      const displayOrder = queryTableOrder();
+
+      const displayedObjects = objectsList.value.filter((object) => {
+        return displayOrder.includes(object.id);
+      });
+
+      displayedObjects.sort((a, b) => {
+        return displayOrder.indexOf(a.id) - displayOrder.indexOf(b.id);
+      });
+      objectStore.setSelectedObjects(displayedObjects);
+      isInitialLoad.value = false;
+    } else {
+      objectStore.setSelectedObjects(
+        objectsList.value.slice(currentPageTop.value, currentPageTop.value + itemsPerPage.value)
+      );
+    }
+  } else {
+    objectStore.setSelectedObjects([]);
+  }
+};
+
+const onPage = (event: any) => {
+  currentPageTop.value = event.first;
+  itemsPerPage.value = event.rows;
+  unselectAllObjects();
+};
+
+const onValueChange = (event: any) => {
```

Comment:
Is there any way we can be more specific about the type of event that will be coming in? Sometimes the `any` keyword is necessary, but that is usually indicative of either a breakdown of typescript, or the actual data structure expectations haven't been thoroughly investigated yet.